pkg/v1/remote/index.go

```diff
@@ -162,16 +162,13 @@ func (r *remoteIndex) childByHash(h v1.Hash) (*Descriptor, error) {
 	return nil, fmt.Errorf("no child with digest %s in index %s", h, r.Ref)
 }
 
-func (r *remoteIndex) childRef(h v1.Hash) (name.Reference, error) {
-	return name.ParseReference(fmt.Sprintf("%s@%s", r.Ref.Context(), h), name.StrictValidation)
+func (r *remoteIndex) childRef(h v1.Hash) name.Reference {
+	return r.Ref.Context().Digest(h.String())
```

Comment:
This method might be simple enough to just inline with a comment.

Reply:
Done.